db/collection_test.go

```diff
@@ -995,8 +1181,8 @@ func TestDeleteInstance(t *testing.T) {
 	}
 
 	// Try to delete again
-	if err = c.Delete(res[0]); err != ErrInstanceNotFound {
-		t.Fatalf("cant't delete non-existent instance")
+	if err = c.Delete(res[0]); err != nil {
```

Comment:
This is up for discussion! I needed to be able to delete an instance that _maybe_ exists. In some cases, a remote client that has local changes but hasn't pushed them all yet might try to delete an instance that doesn't yet exist on the remote. This essentially just silently ignores this case. It seems reasonable: you wanted the instance gone, and it is, it just wasn't deleted by you right now ;)

Reply:
Context: #441